src/main/java/hudson/plugins/sonar/utils/SonarUtils.java

```diff
@@ -130,9 +131,10 @@ private static String extractPatternFromLogs(String pattern, Run<?, ?> build) th
    * Collects as much information as it finds from the sonar analysis in the build and adds it as an action to the build.
    * Even if no information is found, the action is added, marking in the build that a sonar analysis ran. 
    */
-  public static SonarAnalysisAction addBuildInfoTo(Run<?, ?> build, String installationName, boolean skippedIfNoBuild) throws IOException {
+  public static SonarAnalysisAction addBuildInfoTo(Run<?, ?> build, FilePath workspace, String installationName, boolean skippedIfNoBuild)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.lang.InterruptedException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1160)